File: Examples/test-suite/java/template_construct_runme.java

import template_construct.*;

public class template_construct_runme {

  static {
    try {
	System.loadLibrary("template_construct");
    } catch (UnsatisfiedLinkError e) {
      System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e);
      System.exit(1);
    }
  }

  public static void main(String argv[]) {
    Foo_int fi = new Foo_int(0);
    fi.delete();
    Foo_short fs = new Foo_short();
    fs.delete();
    fs = new Foo_short();
    fs.delete();
  }
}

File: Examples/test-suite/kotlin/template_construct_runme.kt
@file:JvmName("template_construct_runme")

import template_construct.*

fun main() {
    try {
        System.loadLibrary("template_construct")
    } catch (e: UnsatisfiedLinkError) {
        System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e)
        kotlin.system.exitProcess(1)
    }

    val fi = Foo_int(0)
    fi.delete()
    var fs = Foo_short()
    fs.delete()
    fs = Foo_short()
    fs.delete()
}
